Gems/WhiteBox/Code/Source/Rendering/Atom/WhiteBoxAtomRenderMesh.cpp

```diff
@@ -256,46 +261,24 @@ namespace WhiteBox
         m_meshFeatureProcessor->SetTransform(m_meshHandle, worldFromLocal);
     }
 
-    void AtomRenderMesh::UpdateMaterial([[maybe_unused]] const WhiteBoxMaterial& material)
+    void AtomRenderMesh::UpdateMaterial(const WhiteBoxMaterial& material)
     {
-        // colors: vertex colors probs not used.
-        // (use constant color for material -> material editor)
-        //
-        // m_meshFeatureProcessor->SetMaterialAssignmentMap() (not in spectra mainline yet)
-        //
-        // for now:
-        // m_meshFeatureProcessor->ReleaseMesh(m_meshHandle);
-        // MaterialAssignmentMap materialMap;
-        // materialMap[DefaultMaterialAssignmentId] = (see below)
-        // m_meshHandle = m_meshFeatureProcessor->AcquireMesh(m_modelAsset, materialMap)
-
-        // from tommy (SkinnedMeshContainer.cpp)
-        /*
-            auto materialAsset =
-           AZ::RPI::AssetUtils::LoadAssetByProductPath<AZ::RPI::MaterialAsset>(SkinnedMeshMaterial); auto
-           materialOverrideInstance = AZ::RPI::Material::FindOrCreate(materialAsset);
-
-            AZ::Render::MaterialAssignmentMap materialMap;
-            materialMap[AZ::Render::DefaultMaterialAssignmentId].m_materialAsset = materialAsset;
-            materialMap[AZ::Render::DefaultMaterialAssignmentId].m_materialInstance = materialOverrideInstance;
-        */
-
-        /*
-            changing material properties:
-
-            (todo)
-        */
+        auto& materialAssignment = m_materialMap[AZ::Render::DefaultMaterialAssignmentId];
+        materialAssignment.m_propertyOverrides[AZ::Name("baseColor.color")] = AZ::Color(material.m_tint);
+        materialAssignment.m_propertyOverrides[AZ::Name("baseColor.useTexture")] = material.m_useTexture;
+        materialAssignment.ApplyProperties();
```

Comment:
You should check the return value of ApplyProperties, depending on when/how frequently this is called. If it's ever called more than once in a frame, it will fail the second time. The MaterialComponentController checks the result, and queues the change for the next frame if it fails the first time.

I'm guessing the whitebox component doesn't get into that situation, so I'm not saying you need to follow the pattern that's in MeshComponentController.cpp. But having an AZ_Error if ApplyProperties fails will at least leave a trail so that we know where to go to fix the issue if that situation comes up in the future.